apiserver/facades/agent/storageprovisioner/register.go

```diff
@@ -63,5 +63,6 @@ func newFacadeV4(stdCtx context.Context, ctx facade.ModelContext) (*StorageProvi
 		serviceFactory.Storage(registry),
 		ctx.Logger().Child("storageprovisioner"),
 		modelInfo.UUID,
+		ctx.ControllerUUID(),
```

Comment:
You can remove the `ControllerConfig` service input, as it's only used to get this.